lang/cs/Org.Apache.REEF.Client/API/IREEFClient.cs

```diff
@@ -50,5 +51,12 @@ public interface IREEFClient
         /// <returns></returns>
         [Unstable("0.14", "Working in progress for rest API status returned")]
         Task<FinalState> GetJobFinalStatus(string appId);
+
+        /// <summary>
+        /// Returns all the application reports running in the cluster
+        /// </summary>
+        /// <returns></returns>
+        [Unstable("0.16", "Working in progress for rest API id returned")]
+        Task<List<IApplicationReport>> GetApplicationReports();
```

Comment:
I am planning to have a new interface and implement couple of more YARN APIs. But not in this iteration though. Introducing a new interface involves in changes in existing tests and introducing more tests. I hope my proposal is okay here.